Set VCC and GND as power inputs in Atmel components

corna opened the pull request:
As reported in the web (like [here](https://forum.kicad.info/t/please-explain-electrical-type-when-create-new-ic-pin/742/3)), VCC and GND should be set as power input, while in the Atmel library they were sometimes set (randomly it seems) as power outputs.

I've not checked all the pins manually (I used a regex to fix the errors), but I suppose that every pin named _VCC_ or _GND_ should be a power input.

diggit commented:
Hi, thanks for fixes!
There is tool for these checkings: checklib.py from [kicad-library-utils](https://github.com/KiCad/kicad-library-utils)

Here is check run on your PR, there are some more pins to fix. Can you look at them too?
It is nicely colored in terminal ;)
There is false alarm on AREF pin, leave it bidirectional.
It is possible to modify EC01 rule to ignore AREF pin and run checklib.py with additional --fix parameter to fix it for you.
```
checklib.py -s -vvv -r EC01 library/atmel.lib
checking component: AT90CAN128-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: AT90CAN128-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: AT90PWM1-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (21): I (INPUT), expected: W (POWER INPUT)
checking component: AT90PWM1-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (19): I (INPUT), expected: W (POWER INPUT)
checking component: AT90S2333-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: AT90S2333-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (21): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA128-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA128-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA1280-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (98): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA1281-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA1281-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA128A-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA128A-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA165-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA165-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA165P-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA165P-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA165PA-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA165PA-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168-MM
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (17): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (21): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168A-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168A-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168A-MM
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (17): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA168A-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (21): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA169-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA169-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA169P-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA169P-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA169PA-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA169PA-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA16U4-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (42): P (PASSIVE), expected: W (POWER INPUT)
checking component: ATMEGA16U4-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (42): P (PASSIVE), expected: W (POWER INPUT)
checking component: ATMEGA325-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA325-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA3250-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (98): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA328PB-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA328PB-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA329-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA329-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA3290-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (98): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA64-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA64-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA64A-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA64A-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (62): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA8-AI
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA8-MI
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATMEGA8-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (21): B (BIDI), expected: W (POWER INPUT)
checking component: ATTINY48-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AVCC (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATXMEGA256A3B-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATXMEGA256A3B-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATXMEGA256A3BU-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATXMEGA256A3BU-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATmega32U4
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (42): I (INPUT), expected: W (POWER INPUT)
    pin GND (23): I (INPUT), expected: W (POWER INPUT)
    pin GND (43): I (INPUT), expected: W (POWER INPUT)
    pin VCC (14): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (24): I (INPUT), expected: W (POWER INPUT)
    pin VCC (34): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (44): I (INPUT), expected: W (POWER INPUT)
    pin GND (15): I (INPUT), expected: W (POWER INPUT)
    pin GND (35): I (INPUT), expected: W (POWER INPUT)
checking component: ATmega32U4-AU
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AREF (42): I (INPUT), expected: W (POWER INPUT)
    pin GND (23): I (INPUT), expected: W (POWER INPUT)
    pin GND (43): I (INPUT), expected: W (POWER INPUT)
    pin VCC (14): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (24): I (INPUT), expected: W (POWER INPUT)
    pin VCC (34): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (44): I (INPUT), expected: W (POWER INPUT)
    pin GND (15): I (INPUT), expected: W (POWER INPUT)
    pin GND (35): I (INPUT), expected: W (POWER INPUT)

```

corna commented:
I've removed the AREF check in rules/EC01.py and run `./checklib.py -s -vvv -r EC01 ~/Desktop/atmel.lib` on the original atmel.lib:

`````
checking component: AT89C2051-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT89C2051-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT89S2051-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT89S2051-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT90PWM1-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (20): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT90PWM1-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (6): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (18): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT90S1200-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT90S1200-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT90S1200-Y
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT90S2313-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: AT90S2313-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY1634-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (8): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY1634-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY1634-U
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (C1): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY2313-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (8): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY2313-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY2313-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY2313A-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY2313A-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY26-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (21): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY26-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY26-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY261-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (21): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY261-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY261-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY261A-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (21): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY261A-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY261A-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY261A-X
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY4313-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY4313-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY43U-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (7): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (8): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY43U-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (9): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
checking component: ATTINY48-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AVCC (20): B (BIDI), expected: W (POWER INPUT)
checking component: ATXMEGA256A3B-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATXMEGA256A3B-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATXMEGA256A3BU-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATXMEGA256A3BU-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
checking component: ATmega32U4
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (23): I (INPUT), expected: W (POWER INPUT)
    pin GND (43): I (INPUT), expected: W (POWER INPUT)
    pin VCC (14): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (24): I (INPUT), expected: W (POWER INPUT)
    pin VCC (34): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (44): I (INPUT), expected: W (POWER INPUT)
    pin GND (15): I (INPUT), expected: W (POWER INPUT)
    pin GND (35): I (INPUT), expected: W (POWER INPUT)
checking component: ATmega32U4-AU
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (23): I (INPUT), expected: W (POWER INPUT)
    pin GND (43): I (INPUT), expected: W (POWER INPUT)
    pin VCC (14): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (24): I (INPUT), expected: W (POWER INPUT)
    pin VCC (34): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (44): I (INPUT), expected: W (POWER INPUT)
    pin GND (15): I (INPUT), expected: W (POWER INPUT)
    pin GND (35): I (INPUT), expected: W (POWER INPUT)
`````

All warnings seems valid to me, so I fixed them with `./checklib.py -s -vvv -r EC01 --fix ~/Desktop/atmel.lib`

`````
checking component: AT89C2051-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT89C2051-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT89S2051-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT89S2051-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT90PWM1-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (20): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT90PWM1-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (6): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (18): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT90S1200-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT90S1200-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT90S1200-Y
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT90S2313-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: AT90S2313-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY1634-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (8): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY1634-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY1634-U
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (C1): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY2313-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (8): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY2313-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY2313-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY2313A-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY2313A-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY26-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (21): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY26-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY26-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY261-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (21): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY261-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY261-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY261A-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (4): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (21): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY261A-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY261A-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY261A-X
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (5): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin AGND (16): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin AGND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY4313-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY4313-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY43U-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (7): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (8): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY43U-S
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VCC (9): w (POWER OUTPUT), expected: W (POWER INPUT)
    pin GND (10): w (POWER OUTPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VCC (w - POWER OUTPUT) to (W - POWER INPUT)
    changing pin GND (w - POWER OUTPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATTINY48-P
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin AVCC (20): B (BIDI), expected: W (POWER INPUT)
    Fixing...
    changing pin AVCC (B - BIDI) to (W - POWER INPUT)
    everything fixed
checking component: ATXMEGA256A3B-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VBAT (I - INPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATXMEGA256A3B-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VBAT (I - INPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATXMEGA256A3BU-A
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VBAT (I - INPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATXMEGA256A3BU-M
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin VBAT (51): I (INPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin VBAT (I - INPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATmega32U4
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (23): I (INPUT), expected: W (POWER INPUT)
    pin GND (43): I (INPUT), expected: W (POWER INPUT)
    pin VCC (14): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (24): I (INPUT), expected: W (POWER INPUT)
    pin VCC (34): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (44): I (INPUT), expected: W (POWER INPUT)
    pin GND (15): I (INPUT), expected: W (POWER INPUT)
    pin GND (35): I (INPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    changing pin VCC (I - INPUT) to (W - POWER INPUT)
    changing pin AVCC (I - INPUT) to (W - POWER INPUT)
    changing pin VCC (I - INPUT) to (W - POWER INPUT)
    changing pin AVCC (I - INPUT) to (W - POWER INPUT)
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    everything fixed
checking component: ATmega32U4-AU
  Violating EC01 - Extra Checking
    Check pins names against pin types.
    pin GND (23): I (INPUT), expected: W (POWER INPUT)
    pin GND (43): I (INPUT), expected: W (POWER INPUT)
    pin VCC (14): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (24): I (INPUT), expected: W (POWER INPUT)
    pin VCC (34): I (INPUT), expected: W (POWER INPUT)
    pin AVCC (44): I (INPUT), expected: W (POWER INPUT)
    pin GND (15): I (INPUT), expected: W (POWER INPUT)
    pin GND (35): I (INPUT), expected: W (POWER INPUT)
    Fixing...
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    changing pin VCC (I - INPUT) to (W - POWER INPUT)
    changing pin AVCC (I - INPUT) to (W - POWER INPUT)
    changing pin VCC (I - INPUT) to (W - POWER INPUT)
    changing pin AVCC (I - INPUT) to (W - POWER INPUT)
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    changing pin GND (I - INPUT) to (W - POWER INPUT)
    everything fixed
`````

Now `./checklib.py -s -vvv -r EC01 ~/Desktop/atmel.lib` doesn't print any warning anymore.

diggit commented:
Thanks for fixes!
